spec.bs

```diff
@@ -3003,6 +3014,49 @@ The <dfn for=Navigator method>getInterestGroupAdAuctionData(|config|)</dfn> meth
 
 </div>
 
+<div algorithm>
+  To <dfn>parse and verify ad auction config</dfn> given an {{AdAuctionDataConfig}} |configIDL|:
+
+  1. Let |config| be a new [=auction data config=] with the following [=struct/items=]:
+    : [=auction data config/seller=]
+    :: The result of running [=parsing an https origin=] on
+       |configIDL|["{{AdAuctionDataConfig/seller}}"]
+    : [=auction data config/coordinator=]
+    :: The result of running [=parsing an https origin=] on
+       |configIDL|["{{AdAuctionDataConfig/coordinatorOrigin}}"]
+    : [=auction data config/request size=]
+    :: |configIDL|["{{AdAuctionDataConfig/requestSize}}"]
+    : [=auction data config/per buyer config=]
+    :: The result of running [=parse per buyer auction data configs=] on
+      |configIDL|["{{AdAuctionDataConfig/perBuyerConfig}}"]
+  1. If |config|'s [=auction data config/seller=] or [=auction data config/coordinator=]
+     fields are failure, then [=exception/throw=] a {{TypeError}}.
+  1. If |config|'s [=auction data config/per buyer config=] [=map/is not empty=]
+     and |config|'s [=auction data config/request size=] is null:
+     1. Let |request_size| equal 0.
+     1. [=list/For each] |buyerConfig| of |config|'s [=auction data config/per buyer config=]'s [=map/values=]:
+        1. If |buyerConfig|'s [=auction data buyer config/size=] is null,

```

Comment:
I'm confused here. {{AdAuctionDataBuyerConfig/targetSize}} is optional, why are we throwing an error when [=auction data buyer config/size=] is null then? When {{AdAuctionDataBuyerConfig/targetSize}} is optional, do we just want to ignore that entry in [=parse per buyer auction data configs=] below? If so, then we can make [=auction data buyer config/size=] not nullable, and only has an entry for it when its IDL value is not null?

Reply:
Target size is only optional if any of the following:
1. [=auction data config/per buyer config=] [=map/is empty=]
2. Every [=auction data config/per buyer config=] specifies a non-null value for  [=auction data buyer config/size=].
3.

Reply:
as discussed offline, all [=auction data buyer config/size=] are required and cannot be null if {{AdAuctionDataBuyerConfig/targetSize}} is null. So resolving this, but added a comment below suggesting to make it clearer about this.